### DISCUSSION TITLE
Excluding languages not working

### DISCUSSION THREAD

**Discussion by igorskyflyer** [github-linguist/linguist#6394] [Classification] [answered]
Hello, I have a [`.gitattributes`](https://github.com/igorskyflyer/ad-void/blob/main/.gitattributes) file with the following contents:

```
*		 text=auto

docs/**  -linguist-documentation linguist-detectable
*.txt    linguist-detectable linguist-language=AdBlock
```

And the language bar being rendered as:

<p align="center"><img src="https://user-images.githubusercontent.com/20957750/235248637-e1d85388-9fd4-417a-beae-b9a256878c92.png" alt="Without svg"></p>

I added the detection for the docs folder since it's ignored by default. Now, what I would like to do is to exclude non-programming files: .svg, .md and .json.

So my `.gitattributes` looks like:

```
*		 text=auto

docs/**  -linguist-documentation linguist-detectable
*.txt    linguist-detectable linguist-language=AdBlock
*.svg    -linguist-detectable
```

The last line should exclude .svg files from the stats but unfortunately that doesn't work. The following is rendered instead:

<p align="center"><img src="https://user-images.githubusercontent.com/20957750/235249078-a2877a7f-374c-4025-9eeb-d625d0167b8b.png" alt="With svg"></p>

Everything else but the Adblock filter and JS is shown as Other. Why is that, is that intended or a bug?

Note: I tried with all 3 extensions (.md, .svg, .json), same thing, this is just an example.

**Comment by lildude** [comment, marked as answer]
> Everything else but the Adblock filter and JS is shown as Other. Why is that, is that intended or a bug?

This is expected. It means there is one or more identified languages that are less than 1% of the total identified languages by bytes of code. In this case, as it's more than 1%, it'll be multiple languages, probably the HTML (~0.3%) + CSS (~1.0%) + something else (JSON? - ~0.3%) depending on what the overrides were when you took the screenshot (I'm AFK so can't check myself).

You might be able to get rid of this by continuing to add more exclusions, or switch things around and only add `docs` overrides for the files you want to count.

Start with just overriding the documentation:

```gitattributes
docs/**  -linguist-documentation
```

This will count all the files in there that are normally counted outside of this directory, which won't include `.md`, `.json` or `.svg`. This may still result in the Other item because the proportion of HTML and CSS and other languages is so low.

**Reply by igorskyflyer** [reply]
Yes - your last paragraph - it seems that's the case, there isn't enough HTML and CSS and that's why it's shown as Other. I'll be committing some new HTML/CSS soon and see if then it will work as expected. Thanks!

**Reply by igorskyflyer** [reply]
Just checked, that was the case, I uploaded a large dummy HTML file and then the results were as expected. 👍🏼
